src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Metadata/PE/IsConstModifierTests.vb

```diff
@@ -0,0 +1,625 @@
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Methods_ReturnTypes_NoModifiers()
+            Dim reference = CreateCSharpCompilation("
+public class TestRef
+{
+    private int value = 0;
+    public ref readonly int M()
+    {
+        return ref value;
+    }
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim obj = New TestRef()
+        obj.M()
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib(source, references:={reference})
+
+            AssertTheseDiagnostics(compilation, <expected>
+BC30657: 'M' has a return type that is not supported or parameter types that are not supported.
+        obj.M()
+            ~
+                                                </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Properties_Virtual()
+            Dim reference = CreateCSharpCompilation("
+public class TestRef
+{
+    private int value = 0;
+    public virtual ref readonly int P => ref value;
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim obj = New TestRef()
+        Dim value = obj.P
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib(source, references:={reference})
+
+            AssertTheseDiagnostics(compilation, <expected>
+BC30643: Property 'P' is of an unsupported type.
+        Dim value = obj.P
+                        ~
+                                                </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Properties_Static()
+            Dim reference = CreateCSharpCompilation("
+public class TestRef
+{
+    private static int value = 0;
+    public static ref readonly int P => ref value;
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim value = TestRef.P
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib(source, references:={reference})
+
+            AssertTheseDiagnostics(compilation, <expected>
+BC30643: Property 'P' is of an unsupported type.
+        Dim value = TestRef.P
+                            ~
+                                                </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Properties_NoModifiers()
```

Comment:
We always place modreqs on ref readonly returns even if there are no modifiers. because consumers might try to change its value.

---
In reply to: [119781166](https://github.com/dotnet/roslyn/pull/19658#discussion_r119781166) [](ancestors = 119781166)